Mathlib/SetTheory/Game/PGame.lean

```diff
@@ -731,6 +732,42 @@ theorem leftResponse_spec {x : PGame} (h : 0 ≤ x) (j : x.RightMoves) :
   Classical.choose_spec <| (zero_le.1 h) j
 #align pgame.left_response_spec SetTheory.PGame.leftResponse_spec
 
+#noalign pgame.upper_bound
+#noalign pgame.upper_bound_right_moves_empty
+#noalign pgame.le_upper_bound
+#noalign pgame.upper_bound_mem_upper_bounds
+
+/-- A small family of pre-games is bounded above. -/
+lemma bddAbove_range_of_small [Small.{u} ι] (f : ι → PGame.{u}) : BddAbove (Set.range f) := by
+  let x : PGame.{u} := ⟨Σ i, (f $ (equivShrink.{u} ι).symm i).LeftMoves, PEmpty,
+    fun x ↦ moveLeft _ x.2, PEmpty.elim⟩
+  refine ⟨x, Set.forall_range_iff.2 fun i ↦ ?_⟩
+  rw [← (equivShrink ι).symm_apply_apply i, le_iff_forall_lf]
+  simpa using fun j ↦ @moveLeft_lf x ⟨equivShrink ι i, j⟩
```

Comment:
On the other hand, maybe this lemma is good for searchability? A user might be expecting to prove `BddAbove` via a family and not know what `Small` means so fail to identify that `bddAbove_of_small` is applicable?

Reply:
Most predicates in mathlib come in two kinds: A set version and an indexed version. You can define `IsFooSet s` as `IsFooFamily (Subtype.val : s → α)` and you can define `IsFooFamily f` as `IsFooSet (Set.range f)` (assuming duplicates in your family don't matter). However, it is usually impractical to use one predicate in place of the other:
* When using `IsFooFamily (Subtype.val : s → α)`, elements of `s` are now images of the coercion `Subtype.val : s → α`
* When using `IsFooSet (Set.range f)`, you need to replace `y ∈ Set.range f` by `f x`, `x` and vice-versa.

It is therefore preferrable to have both predicates (see eg `IsDirected`). This is currently not the case for `BddAbove`, where we only have the set version. Hence I am acting as if `BddAbove (Set.range _)` was a predicate, because many lemmas expect it in that form and because it might become one in the future. The fact that `bddAbove_range_of_small` can be syntactically derived from `bddAbove_of_small` is somehow an accident of us not having the family predicate, and I would rather not muddle the waters by only having the set version of the lemma.

Reply:
Interesting! Thanks for the explanation